sqs/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/SqsSink.scala

```diff
@@ -0,0 +1,335 @@
+    @volatile private var lastFlushedTime = 0L
+
+    def store(event: Array[Byte]): Unit = {
+      val eventBytes = ByteBuffer.wrap(event)
+      val eventSize  = eventBytes.capacity
+      if (eventSize >= MaxBytes) {
+        log.error(
+          s"Message of size $eventSize bytes is too large. It must be less than $MaxBytes bytes."
+        )
+      } else {
+        synchronized {
+          storedEvents = eventBytes :: storedEvents
+          byteCount += eventSize
+          if (storedEvents.size >= RecordThreshold || byteCount >= ByteThreshold) {
+            flush()
+          }
+        }
+      }
+    }
+
+    def flush(): Unit = {
+      val eventsToSend = synchronized {
+        val evts = storedEvents.reverse
+        storedEvents = Nil
+        byteCount    = 0
+        evts
+      }
+      lastFlushedTime = System.currentTimeMillis()
+      sendBatch(eventsToSend)
+    }
+
+    def getLastFlushTime(): Long = lastFlushedTime
+
+    /**
+      * Recursively schedule a task to flush the EventBuffer.
+      * Even if the incoming event flow dries up, all buffered events will eventually get sent.
+      * Whenever TimeThreshold milliseconds have passed since the last call to flush, call flush.
+      * @param interval When to schedule the next flush
+      */
+    def scheduleFlush(interval: Long = TimeThreshold): Unit = {
+      executorService.schedule(
+        new Thread {
+          override def run(): Unit = {
+            val lastFlushed = getLastFlushTime()
+            val currentTime = System.currentTimeMillis()
+            if (currentTime - lastFlushed >= TimeThreshold) {
+              flush()
+              scheduleFlush(TimeThreshold)
+            } else {
+              scheduleFlush(TimeThreshold + lastFlushed - currentTime)
+            }
+          }
+        },
+        interval,
+        MILLISECONDS
+      )
+      ()
+    }
+  }
+
+  def sendBatch(batch: List[EventBuffer.Event], nextBackoff: Long = minBackoff): Unit =
+    if (batch.nonEmpty) {
+      log.info(s"Writing ${batch.size} Thrift records to SQS queue ${queueName}")
+
+      multiPut(batch).onComplete {
+        case Success(s) =>
+          log.info(s"Successfully wrote ${batch.size - s.size} out of ${batch.size} messages.")
+          if (s.nonEmpty) {
+            s.foreach { f =>
+              log.error(
+                s"Message failed with error code [${f._2.code}] and message [${f._2.message}]"
+              )
+            }
+            log.error(s"Retrying in $nextBackoff milliseconds...")
+            scheduleBatch(s.keys.toList, nextBackoff)
+          }
+        case Failure(f) =>
+          log.error("Writing failed.", f)
+          log.error(s"Retrying in $nextBackoff milliseconds...")
+          scheduleBatch(batch, nextBackoff)
+      }
+    }
+
+  // The batch request can return unsuccessful entries. In this case, we return a map of Events to be retried and the reasons they failed to be inserted.
```

Comment:
```suggestion
/** @return Empty map in case of success, non-empty map of Events to be retried and the reasons why they failed to be inserted in case of problem. */
```